src/session.rs

```diff
@@ -49,18 +49,46 @@ pub struct Cookie {
     secure: Option<bool>,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+struct Header {
+    key: String,
+    value: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+enum Headers {
+    Map(HashMap<String, String>),
+    List(Vec<Header>),
```

Comment:
Can you add a note explaining that this is to support duplicate keys, and that we keep `Map` around for backward compatibility?